### ISSUE TITLE
open source code

### ISSUE THREAD

**Issue by ShiyuCHEN2003** [NewCodeAlg/DiffusionForFaultDiagnosis#1]
Hi, I've seen your article on TII has been published in May, when will the source code be open please?

**Comment by NewCodeAlg** [comment]
Thank you for your reminder. The network model code  has been uploaded now.

**Comment by ShiyuCHEN2003** [comment]
Thank you so much!!!  I still have a small question, in the paper mentioned case1: when using CWRU dataset, there are 400 segments for each type, and the length of each segment is 512, but the official dataset such as X186_DE_time variable has only 122,136 data points, which is less than 512*400, is this part to do the overlap operation to extend it to 400 segments?

**Comment by NewCodeAlg** [comment]
Yes, your understanding is correct. We used overlapping sliding windows

**Comment by ShiyuCHEN2003** [comment]
hello, I'm sorry that I have some problems again. When I used the original dataset and the generated dataset for fault diagnosis, the accuracy rate was relatively low. After I printed out the wrong samples, I found that it was difficult to distinguish the three fault types of ball fault with a diameter of 7, 14 and 21, resulting in a low accuracy rate. This problem will occur even if I use all the original dataset (no data is generated). I use Lenet and Alexnet for classifiers. Do you have any suggestions？
![1721129416793](https://github.com/user-attachments/assets/2775f72a-2a3

**Comment by ShiyuCHEN2003** [comment]
hello, I'm sorry that I have some problems again. When I used the original dataset and the generated dataset for fault diagnosis, the accuracy rate was relatively low. After I printed out the wrong samples, I found that it was difficult to distinguish the three fault types of ball fault with a diameter of 7, 14 and 21, resulting in a low accuracy rate. This problem will occur even if I use all the original dataset (no data is generated). I use Lenet and Alexnet for classifiers. Do you have any suggestions？
![1721129416793](https://github.com/user-attachments/assets/c173b344-ad42-4442-af3c-819a80360bea)

**Comment by NewCodeAlg** [comment]
Please confirm that the sample image you saved is in PNG format. If using JPG format, there may be compression issues. For classification networks, you can use ResNet.

**Comment by ShiyuCHEN2003** [comment]
Thank you for your prompt response. I have noted that the PNG format is being utilized. I am planning to experiment with the ResNet-18 network for this project.   hahaha~
